retriever/bin/retriever.js

```diff
@@ -17,56 +18,63 @@ export default {
    * @returns
    */
   async fetch(request, env, ctx, { retrieveFile = defaultRetrieveFile } = {}) {
+    try {
+      return await this._fetch(request, env, ctx, retrieveFile)
+    } catch (error) {
+      console.error(error)
+      const message =
+        typeof error === 'object' &&
+        error !== null &&
+        'message' in error &&
+        typeof error.message === 'string'
+          ? error.message
+          : String(error)
+      const status =
+        typeof error === 'object' &&
+        error !== null &&
+        'status' in error &&
+        typeof error.status === 'number'
+          ? error.status
+          : 500
+      return new Response(message, { status })
+    }
+  },
+
+  /**
+   * @param {Request} request
+   * @param {Env} env
+   * @param {ExecutionContext} ctx
+   * @param {typeof defaultRetrieveFile} retrieveFile
+   * @returns
+   */
+  async _fetch(request, env, ctx, retrieveFile) {
     const requestTimestamp = new Date().toISOString()
     const workerStartedAt = performance.now()
     const requestCountryCode = request.headers.get('CF-IPCountry')
-    if (request.method !== 'GET') {
-      return new Response('Method Not Allowed', { status: 405 })
-    }
+    httpAssert(request.method === 'GET', 405, 'Method Not Allowed')
 
-    const {
-      clientWalletAddress,
-      rootCid,
-      error: parsingError,
-    } = parseRequest(request, env)
-    if (parsingError) {
-      console.error(parsingError)
-      return new Response(parsingError, { status: 400 })
-    }
+    const { clientWalletAddress, rootCid } = parseRequest(request, env)
```

Comment:
@bajtos through forwarding errors (from `_fetch` to `fetch`), we don't need to write a `try/catch` for every call. This is the main lesson from using something like `http-assert` and then having a global error handling (here: `fetch`)